File: backend/src/application/schemas/agent.py
from pydantic import BaseModel
from typing import List, Dict, Any
from datetime import datetime

class AgentRuntimeDTO(BaseModel):
    id: str
    status: str
    current_task: str | None
    uptime: int

class AgentSessionDTO(BaseModel):
    session_id: str
    started_at: datetime
    duration: int

class AgentThoughtDTO(BaseModel):
    thought_id: str
    timestamp: datetime
    content: str
    confidence: float

class AgentActionDTO(BaseModel):
    action_id: str
    timestamp: datetime
    tool_name: str
    input: Dict[str, Any]
    result: str

class AgentToolDTO(BaseModel):
    tool_name: str
    description: str
    usage_count: int

class AgentMemoryDTO(BaseModel):
    memory_id: str
    content: str
    relevance: float

class AgentReflectionDTO(BaseModel):
    reflection_id: str
    timestamp: datetime
    insight: str

class AgentReplayDTO(BaseModel):
    events: List[Dict[str, Any]]

File: frontend/src/types/agent.ts
import { StandardResponse } from "./dashboard";

export interface AgentRuntime {
  id: string;
  status: string;
  current_task: string | null;
  uptime: number;
}

export interface AgentSession {
  session_id: string;
  started_at: string;
  duration: number;
}

export interface AgentThought {
  thought_id: string;
  timestamp: string;
  content: string;
  confidence: number;
}

export interface AgentAction {
  action_id: string;
  timestamp: string;
  tool_name: string;
  input: Record<string, any>;
  result: string;
}

export interface AgentTool {
  tool_name: string;
  description: string;
  usage_count: number;
}

export interface AgentMemory {
  memory_id: string;
  content: string;
  relevance: number;
}

export interface AgentReflection {
  reflection_id: string;
  timestamp: string;
  insight: string;
}

export interface AgentReplay {
  events: Record<string, any>[];
}
